Parser.fs

```diff
@@ -148,20 +157,26 @@ let parseSymbolic =
 
 /// Parser for primary expressions.
 let parsePrimaryExpression =
-    choice [ pstring "true" >>% True
-             pstring "false" >>% False
-             pint64 |>> Int
-             parseLValue |>> LV
-             parseSymbolic |>> Symbolic
-             inParens parseExpression ] .>> ws
+    let expressions =  
+        (inParens parseExpression) :: List.map nodify [ 
+            pstring "true" >>% True
+            pstring "false" >>% False
+            pint64 |>> Int
+            parseLValue |>> LV
+            parseSymbolic |>> Symbolic
+    ]
+
+    choice expressions .>> ws
 
 /// Generic parser for tiers of binary expressions.
 /// Accepts the next precedence level parser, and a list of pairs of operator and AST representation.
 /// This generates a LEFT-associative precedence level.
 let parseBinaryExpressionLevel nextLevel expList =
+    let parse_bop (ops, op) = nodify (pstring ops) .>> ws |>> fun x -> fun a b -> { Node = Bop(op, a, b); Position = x.Position }
```

Comment:
Could move this out into a separate function, I'm not sure but I think this might end up creating a new function every time you run `parseBinaryExpressionLevel`